astropy/table/table.py

```diff
@@ -3021,8 +3050,80 @@ def __eq__(self, other):
 
         return result
 
-    def __ne__(self, other):
-        return ~self.__eq__(other)
+    def values_equal(self, other):
+        """
+        Element-wise comparison of table with another table, list, or scalar.
+
+        Returns a ``Table`` with the same columns containing boolean values
+        showing result of comparison.
+
+        Parameters
+        ----------
+        other : Table-like object or list or scalar
+             Object to compare with table
+
+        Examples
+        --------
+        Compare one Table with other::
+
+          >>> t1 = Table([[1, 2], [4, 5], [-7, 8]], names=('a', 'b', 'c'))
+          >>> t2 = Table([[1, 2], [-4, 5], [7, 8]], names=('a', 'b', 'c'))
+          >>> t1.values_equal(t2)
+          <Table length=2>
+           a     b     c
+          bool  bool  bool
+          ---- ----- -----
+          True False False
+          True  True  True
+
+        """
+        if isinstance(other, Table):
+            names = other.colnames
+        else:
+            try:
+                other = Table(other, copy=False)
+                names = other.colnames
+            except Exception:
+                # Broadcast other into a dict, so e.g. other = 2 will turn into
+                # other = {'a': 2, 'b': 2} and then equality does a
+                # column-by-column broadcasting.
+                names = self.colnames
+                other = {name: other for name in names}
+
+        # Require column names match but do not require same column order
+        if set(self.colnames) != set(names):
+            raise ValueError('cannot compare tables with different column names')
+
+        eqs = []
+        for name in names:
+            try:
+                np.broadcast(self[name], other[name])  # Check if broadcast-able
```

Comment:
I thought this would fail on `Time`, but it is actually smarter than that, neat!